### PR TITLE
Bookstack: Fix Update function composer

### PR DESCRIPTION
**Pull request by michelroegl-brunner**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description

In the preveious version of this script the update would fail with 
```
✓ Services Stopped / Updating Bookstack to 24.10.3    [ERROR] in line 73: exit code 0: while executing command COMPOSER_ALLOW_SUPERUSER=1 composer install --no-dev &> /dev/null
```
Also would the script stop at **php artisan key:generate ** and ** php artisan migrate** without the switch **--force**

## Type of change
Please check the relevant option(s):

- [X] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [X] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [X] Testing performed (I have tested my changes, ensuring everything works as expected)
- [ ] Documentation updated (I have updated any relevant documentation)

## Additional Information (optional)
Provide any additional context or screenshots about the feature or fix here.

Fixes the problem in the [Discusion](https://github.com/community-scripts/ProxmoxVE/discussions/582)

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
- Related PR #

### CODE REVIEW

**Review by MickLesk — DISMISSED** [review]

File: ct/bookstack.sh [hunk 1]
```diff
@@ -70,9 +70,10 @@ if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_v
   unzip -q v${RELEASE}.zip
   mv BookStack-${RELEASE} /opt/bookstack
   mv /opt/.env /opt/bookstack/.env
-  COMPOSER_ALLOW_SUPERUSER=1 composer install --no-dev  &>/dev/null
-  php artisan key:generate &>/dev/null
-  php artisan migrate &>/dev/null
+  cd /opt/bookstack
+  composer install --no-dev --no-interaction &>/dev/null
```
Comment (havardthom) [inline ct/bookstack.sh:74]: Why remove `COMPOSER_ALLOW_SUPERUSER=1`? What was the error message before?

Comment (michelroegl-brunner) [reply ct/bookstack.sh:74]: COMPOSER_ALLOW_SUPERUSER=1 is not needed with **--no-interaction**

If failed at this `COMPOSER_ALLOW_SUPERUSER=1 composer install --no-dev &> /dev/null`

In my opinion, the update was never working correctly as it was set up bevor.

Comment (havardthom) [inline ct/bookstack.sh:74]:
```suggestion
  COMPOSER_ALLOW_SUPERUSER=1 composer install --no-dev &>/dev/null
```

This was the original error
```
root@bookstack:~#  COMPOSER_ALLOW_SUPERUSER=1 composer install --no-dev
Composer could not find a composer.json file in /root
To initialize a project, please create a composer.json file. See https://getcomposer.org/basic-usage
```
adding `cd /opt/bookstack` fixes it and the original command runs fine after that.

We should keep `COMPOSER_ALLOW_SUPERUSER=1` to indicate that we are running the command as `root` https://github.com/composer/composer/blob/666dc93fcc1090efffee82d78e4553caca682087/doc/faqs/how-to-install-untrusted-packages-safely.md?plain=1#L13C19-L13C43

Comment (michelroegl-brunner) [reply ct/bookstack.sh]: Fine by me, i´ve changed it.

**Review by MickLesk — APPROVED** [review]

**Review by havardthom — APPROVED** [review]

**Comment by Hindin81** [comment]
Now the update works but after that Bookstack is dead:  

CleanShot 2024-12-08 at 09 18 59@2x

**Comment by michelroegl-brunner** [comment]
Have you just run the script in the lxc shell or did you do anything beforhand? (Testing to update or something like that?)

**Comment by Hindin81** [comment]
No, before start the Update Script in the LXC Container I do a snapshoot of the Container. So when testing it is always the last working State. I installed the Container 3 weeks ago with the latest Community Script and importet the DB and files.

**Pull request merged**